### DISCUSSION TITLE
Subclipse / Team and single sign on

### DISCUSSION THREAD

**Discussion by dbarrieu** [subclipse/subclipse#155] [Q&A]
Hi,

I try to find a way to optimize the authentification to multiple SVN repositories.
For the usage of my Eclipse RCP product, users are synchronize with more than 10 différents SVN repositories. Each of then display a specific authentification so for the first connection, or for a password update, users must put his authentification as many times as there are SVN repository.
Do you know if there actualy a way, in subclipse or Team, to optimise this ? 
Maybe with the usage of the security storage configuration of Eclipse...

Regards.

**Comment by nikopp** [comment]
Hi,

if you use svnserve, you can use the same authentication realm for all
repositories in svnserve.conf:

### This option specifies the authentication realm of the repository.
### If two repositories have the same authentication realm, they should
### have the same password database, and vice versa.  The default realm
### is repository's uuid.
realm = oneForAll

Because the password is locally stored per realm instead of per repo.

Niko

**Reply by dbarrieu** [reply]
Thanks for your quick reply. 

Regards.

**Comment by markphip** [comment]
Subclipse itself does not collect or cache any credentials. Instead it is all done by the SVN API implementation. Using JavaHL, as example, it caches passwords the same as the SVN command line client does and is controlled by the same settings etc. Even the dialog to provide username and password is triggered by the API.  Subclipse provides the API with a callback method to collect credentials and the API decides when and if to call this method when it decides it needs to ask the user for credentials.

SVN's cache is based on the "realm". So if all of your repositories are on the same server and the authentication is all part of the same realm, the user should only be challenged for credentials one time .. provided that the user is allowed to cache their credentials. This is determined by the SVN client configuration.
